src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtSpoeAreasOfLawService.java

```diff
@@ -0,0 +1,106 @@
+package uk.gov.hmcts.dts.fact.services.admin;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.entity.CourtAreaOfLawSpoe;
+import uk.gov.hmcts.dts.fact.exception.DuplicatedListItemException;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw;
+import uk.gov.hmcts.dts.fact.repositories.CourtAreaOfLawSpoeRepository;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+import uk.gov.hmcts.dts.fact.util.AuditType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class AdminCourtSpoeAreasOfLawService {
+
+    private final CourtRepository courtRepository;
+    private final AdminAuditService adminAuditService;
+    private final CourtAreaOfLawSpoeRepository courtAreaOfLawSpoeRepository;
+
+    @Autowired
+    public AdminCourtSpoeAreasOfLawService(final CourtRepository courtRepository, final CourtAreaOfLawSpoeRepository courtAreaOfLawSpoeRepository, final AdminAuditService adminAuditService) {
+        this.courtRepository = courtRepository;
+        this.courtAreaOfLawSpoeRepository = courtAreaOfLawSpoeRepository;
+        this.adminAuditService = adminAuditService;
+    }
+
+    public List<AreaOfLaw> getAllSpoeAreasOfLaw() {
+        return courtAreaOfLawSpoeRepository.findAll()
+            .stream()
+            .map(CourtAreaOfLawSpoe::getAreaOfLaw).distinct()
+            .map(AreaOfLaw::new)
+            .collect(toList());
+    }
+
+    public List<AreaOfLaw> getCourtSpoeAreasOfLawBySlug(final String slug) {
+
+        Court court = courtRepository.findBySlug(slug).orElseThrow(() -> new NotFoundException(slug));
+        return  courtAreaOfLawSpoeRepository.getAllByCourtId(court.getId())
+            .stream()
+            .map(CourtAreaOfLawSpoe::getAreaOfLaw)
+            .map(aol -> new AreaOfLaw(aol,true))
+            .collect(toList());
+    }
+
+    @Transactional()
+    public List<AreaOfLaw> updateSpoeAreasOfLawForCourt(final String slug, final List<AreaOfLaw> areasOfLaw) {
```

Comment:
This all looks great! Just wondering if we can expand on it a bit by either: 

1) getCourtSpoeAreasOfLawBySlug will return null values for the other fields that you do not need, which is perfectly fine given the scope of this. However, can we always be absolutely certain that the data from the returned list will not be used in a put request moving forwards? 

A consideration would be to simply return the values in the model so that they are not null, i.e keep the data consistent. 

I mention this because its fine in this case, but I have a slight worry about if we do this more frequently for other endpoints moving forwards

2) Would this be better off having its own model to avoid nulls being returned?

Reply:
I agree would be much better to separate the model out which I have done now and created 
public class SpoeAreaOfLaw {

    private Integer id;
    private String name;
    private boolean singlePointEntry;

    public SpoeAreaOfLaw(uk.gov.hmcts.dts.fact.entity.AreaOfLaw areaOfLaw) {
        this.id = areaOfLaw.getId();
        this.name = areaOfLaw.getName();
        this.singlePointEntry = true;

    }